.github/workflows/preflight-release.yml

```diff
@@ -46,8 +46,7 @@ jobs:
         env:
           PREFLIGHT_LATEST: ${{ steps.check_release.outputs.preflight_latest }}
         run: |
-          git ls-remote --exit-code --heads origin preflight-release-$PREFLIGHT_LATEST
-          if [ $? -eq 0 ]; then
+          if [ ! git ls-remote --heads origin preflight-release-$PREFLIGHT_LATEST ]; then
```

Comment:
This won't work: without `--exit-code` the command will always return 0:
```
[ kad@kyoraku ansible-collection-redhatci-ocp (✨ rhoai)] ~> git ls-remote --exit-code --heads origin asdfasdf
[ kad@kyoraku ansible-collection-redhatci-ocp (✨ rhoai)] ~> echo $?
2
[ kad@kyoraku ansible-collection-redhatci-ocp (✨ rhoai)] ~> git ls-remote --heads origin asdfasdf
[ kad@kyoraku ansible-collection-redhatci-ocp (✨ rhoai)] ~> echo $?
0
```
and your conditional relies on the return code. You could do something like this instead:
```
if [ "$(git ls-remote --heads origin preflight-release-$PREFLIGHT_LATEST)" == "" ]; then
...
fi
```

**or**

```
exists="yes"
git ls-remote --exit-code --heads origin preflight-release-$PREFLIGHT_LATEST || exists="no"
if [ $exists == "no" ]; then
...
fi
```

Reply:
thanks @thekad agree, sorry for the overlooking the [failing test that looked like green](https://github.com/redhatci/ansible-collection-redhatci-ocp/actions/runs/9418549406/job/25946275862).

Reply:
the issue is fixed, the workflow is [running fine in the case of the existing branch](https://github.com/redhatci/ansible-collection-redhatci-ocp/actions/runs/9468150180/job/26083847662).
```
Run branch_exists=$(git ls-remote --heads origin preflight-release-$PREFLIGHT_LATEST)
  branch_exists=$(git ls-remote --heads origin preflight-release-$PREFLIGHT_LATEST)
  if [ -n "$branch_exists" ]; then
    echo "Branch preflight-release-$PREFLIGHT_LATEST already exists"
    echo "branch_exists=true" >> $GITHUB_OUTPUT
  else
    echo "Branch preflight-release-$PREFLIGHT_LATEST does not exist"
    echo "branch_exists=false" >> $GITHUB_OUTPUT
  fi
  shell: /usr/bin/bash -e {0}
  env:
    PREFLIGHT_LATEST: 1.9.[7](https://github.com/redhatci/ansible-collection-redhatci-ocp/actions/runs/9468150180/job/26083847662#step:6:7)
Branch preflight-release-1.9.7 already exists
```